### PR TITLE
Basic infrastructure for FPGA Manager binding

### PR DESCRIPTION
**Pull request by sonals**

### CODE REVIEW

File: src/runtime_src/driver/xclng/drm/xocl/subdev/fmgr.c [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+
+	/* Check if we got the complete xclbin */
+	if (obj->blob->m_header.m_length != obj->count) {
+		obj->state = FPGA_MGR_STATE_WRITE_COMPLETE_ERR;
+		return -EINVAL;
+	}
+	xocl_info(&mgr->dev, "Finish download of xclbin %pUb of size %zu B", &obj->blob->m_header.uuid, obj->count);
+	obj->state = FPGA_MGR_STATE_WRITE_COMPLETE;
+
+	/* Send the xclbin blob to actual download framework in icap */
```
Comment (maxzhen) [inline src/runtime_src/driver/xclng/drm/xocl/subdev/fmgr.c]: Could  you just call the icap API to really download it (and remove the TODO)? The existing API should work with a kernel virtual address on Intel CPU. At the same time, I'll modify the API to accept a KVA in a separate PR. But should have no impact on any existing consumers including fpga mgr.

Comment (sonals) [reply src/runtime_src/driver/xclng/drm/xocl/subdev/fmgr.c]: Done

**Comment by sonals** [comment]
retest this please

**Comment by maxzhen** [comment]
retest this please

**Comment by premsair** [comment]
retest this please

**Comment by maxzhen** [comment]
retest this please

**Comment by maxzhen** [comment]
@sonals the driver failed to compile, could you check the pipeline log and fix it? the fpga mgr api seems evolving as kernel changes...

**Comment by sonals** [comment]
retest this please

**Pull request merged**